### ISSUE TITLE
Support Postgres uuid functions 

### ISSUE THREAD

**Issue by sjwiesman** [MaterializeInc/materialize#16554]
### Feature request

https://www.postgresql.org/docs/15/uuid-ossp.html

**Comment by aljoscha** [comment]
Can we ever support functions like this, which are not pure/deterministic functions of their input? I don't think we can.

**Comment by sjwiesman** [comment]
Probably not!

**Comment by rjnn** [comment]
We can support `uuid_generate_v5` from the `uuid_ossp` extension, which is a purely deterministic uuid function. The burden is on the user to provide the random input.

**Comment by sjwiesman** [comment]
I'm actively working with someone who could use that function and would be able to seed it with a hash

**Comment by aljoscha** [comment]
@dseisun-materialize This might be the next one to pick off after #15793

**Comment by rjnn** [comment]
That's right, you would need to support the `uuid_ns_*` functions (which all return constants) for `uuid_generate_v5()` to work.

**Comment by rjnn** [comment]
We'd also have to document why we only support `v5` (technically we could also support `v3` although md5 is a pretty sus hash function) and guide people to it in our docs.

**Comment by aljoscha** [comment]
@dseisun-materialize did you create a follow-up issue for docs?

**Comment by dseisun-materialize** [comment]
I didn't. I updated docs in the PR though, or are there other docs you were thinking of?

**Comment by morsapaes** [comment]
We try to keep the implementation and docs PRs separate so that changes aren't surfaced to users ahead of the actual release (which can be confusing (and has been, in the past!)).
